src/assets/rewardrequestdb.cpp

```diff
@@ -0,0 +1,282 @@
+
+        return succeeded;
+    }
+
+    //  Otherwise, erase the entire entry since none are left.
+    bool succeeded = Erase(std::make_pair(SCHEDULEDREWARD_FLAG, rewardToRemove.heightForPayout), true);
+
+    LogPrintf("%s : Removal of last scheduled reward %s!\n",
+        __func__,
+        succeeded ? "succeeded" : "failed");
+
+    return succeeded;
+}
+
+bool CRewardRequestDB::AreRewardsScheduledForHeight(const int & p_maxBlockHeight)
+{
+    LogPrintf("%s : Looking for scheduled rewards at height %d!\n",
+        __func__, p_maxBlockHeight);
+
+    std::unique_ptr<CDBIterator> pcursor(NewIterator());
+
+    pcursor->SeekToFirst();
+
+    // Find out if any pending rewards exist at or below the specified height
+    while (pcursor->Valid()) {
+        boost::this_thread::interruption_point();
+        std::pair<char, int> key;
+
+        //  Only retrieve entries earlier than the provided block height
+        if (pcursor->GetKey(key) && key.first == SCHEDULEDREWARD_FLAG && key.second == p_maxBlockHeight) {
+            return true;
+        }
+
+        pcursor->Next();
+    }
+
+    return false;
+}
+
+bool CRewardRequestDB::LoadPayableRewardsForAsset(
+    const std::string & p_assetName, const int & p_blockHeight,
+    std::set<CRewardRequest> & p_rewardRequests)
+{
+    LogPrintf("%s : Looking for scheduled rewards for asset '%s' at height %d!\n",
+        __func__, p_assetName.c_str(), p_blockHeight);
+
+    std::unique_ptr<CDBIterator> pcursor(NewIterator());
+
+    pcursor->SeekToFirst();
+
+    // Load all pending rewards
+    while (pcursor->Valid()) {
+        boost::this_thread::interruption_point();
+        std::pair<char, int> key;
+
+        //  Only retrieve entries at the provided block height
+        if (pcursor->GetKey(key) && key.first == SCHEDULEDREWARD_FLAG && key.second == p_blockHeight) {
+            CRewardRequestDBEntry reqDbEntry;
+
+            if (pcursor->GetValue(reqDbEntry)) {
+                for (auto const & entry : reqDbEntry.requests) {
+                    //  If an asset was specified, only add entries for it.
+                    //  Otherwise, retrieve all entries.
+                    if (p_assetName.length() == 0 || p_assetName == entry.tgtAssetName) {
+                        p_rewardRequests.insert(entry);
+                    }
+                }
+            } else {
+                LogPrintf("%s: Failed to read reward\n", __func__);
+            }
+        }
+
+        pcursor->Next();
+    }
+
+    for (auto const & reward : p_rewardRequests) {
+        LogPrintf("%s : Found payable reward: rewardID='%s', wallet='%s', height=%d, amt=%lld, srcAsset='%s', tgtAsset='%s', exceptions='%s'\n",
+            __func__,
+            reward.rewardID.c_str(),
+            reward.walletName.c_str(), reward.heightForPayout, static_cast<long long>(reward.totalPayoutAmt),
+            reward.payoutSrc.c_str(), reward.tgtAssetName.c_str(),
+            reward.exceptionAddresses.c_str());
+    }
```

Comment:
//TODO Make sure this doesn't get passed testing
We should create a LogPrint enum for rewards and use that to post to the debug log

Reply:
This block has been removed/changed and now only lists block height and asset name.